### DISCUSSION TITLE
Error when I compile the project. I need help.

### DISCUSSION THREAD

**Discussion by sbatyrov** [eclipse-birt/birt#1998] [Q&A]
I downloaded the project and am trying to build it. I get the following error.

```
[ERROR] Failed to execute goal [32morg.apache.maven.plugins:maven-antrun-plugin:3.1.0:run[m [1m(build-api-package)[m on project [36morg.eclipse.birt.api[m: [1;31mAn Ant BuildException has occured: The following error occurred while executing this line:[m
[ERROR] [1;31m/home/suleiman/git/birt/build/org.eclipse.birt.api/build.xml:30: Manifest file: /home/suleiman/git/birt/build/org.eclipse.birt.api/target/manifest.mf does not exist.[m
[ERROR] [1;31maround Ant part ...<ant antfile="build.xml" dir="/home/suleiman/git/birt/build/org.eclipse.birt.api" target="package">... @ 4:104 in /home/suleiman/git/birt/build/org.eclipse.birt.api/target/antrun/build-main.xml[m
[ERROR] -> [1m[Help 1]
```

I checked, the manifest is created if you manually run ant in the directory: org.eclipse.birt.api 
But when I run the build of the entire project via maven, I get this error: What is the problem?

**Comment by sbatyrov** [comment]
some add info about error 

[DEBUG]      [java] running CreateManifest with default permissions (exit forbidden)
[DEBUG]      [java] Running in same VM Executing 'CreateManifest'
     [java] The ' characters around the executable and arguments are
     [java] not part of the command.
[DEBUG] force loading CreateManifest
[DEBUG] Finding class CreateManifest
[DEBUG] Loaded from /home/suleiman/git/birt/build/org.eclipse.birt.api CreateManifest.class
[DEBUG] Class java.lang.Object loaded from parent loader (parentFirst)
[DEBUG] Class java.util.Collection loaded from parent loader (parentFirst)
[DEBUG] Class java.util.List loaded from parent loader (parentFirst)
[DEBUG] Class java.io.Writer loaded from parent loader (parentFirst)
[DEBUG] Class java.io.FileWriter loaded from parent loader (parentFirst)
[DEBUG] Class java.io.BufferedWriter loaded from parent loader (parentFirst)
[DEBUG] Class java.lang.String loaded from parent loader (parentFirst)
[DEBUG] Class java.io.IOException loaded from parent loader (parentFirst)
[ERROR]      [java] java.lang.UnsupportedOperationException: The Security Manager is deprecated and will be removed in a future release
     [java] 	at org.apache.tools.ant.taskdefs.ExecuteJava.execute(ExecuteJava.java:194)
     [java] 	at org.apache.tools.ant.taskdefs.Java.run(Java.java:891)
     [java] 	at org.apache.tools.ant.taskdefs.Java.executeJava(Java.java:231)
     [java] 	at org.apache.tools.ant.taskdefs.Java.executeJava(Java.java:135)
     [java] 	at org.apache.tools.ant.taskdefs.Java.execute(Java.java:108)
     [java] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
     [java] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
     [java] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
     [java] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
     [java] 	at org.apache.tools.ant.Task.perform(Task.java:350)
     [java] 	at org.apache.tools.ant.Target.execute(Target.java:449)
     [java] 	at org.apache.tools.ant.Target.performTasks(Target.java:470)
     [java] 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1401)
     [java] 	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:36)
     [java] 	at org.apache.tools.ant.Project.executeTargets(Project.java:1264)
     [java] 	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:437)
     [java] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
     [java] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
     [java] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
     [java] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
     [java] 	at org.apache.tools.ant.Task.perform(Task.java:350)
     [java] 	at org.apache.tools.ant.Target.execute(Target.java:449)
     [java] 	at org.apache.tools.ant.Target.performTasks(Target.java:470)
     [java] 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1401)
     [java] 	at org.apache.tools.ant.Project.executeTarget(Project.java:1374)
     [java] 	at org.apache.maven.plugins.antrun.AntRunMojo.execute(AntRunMojo.java:297)
     [java] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
     [java] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
     [java] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
     [java] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
     [java] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
     [java] 	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
     [java] 	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
     [java] 	at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
     [java] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
     [java] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
     [java] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
     [java] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
     [java] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
     [java] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
     [java] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
     [java] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
     [java] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:906)
     [java] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
     [java] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
     [java] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
     [java] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
     [java] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:255)
     [java] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:201)
     [java] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
     [java] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
     [java] Caused by: java.lang.UnsupportedOperationException: The Security Manager is deprecated and will be removed in a future release
     [java] 	at java.base/java.lang.System.setSecurityManager(System.java:430)
     [java] 	at org.apache.tools.ant.types.Permissions.setSecurityManager(Permissions.java:103)
     [java] 	at org.apache.tools.ant.taskdefs.ExecuteJava.run(ExecuteJava.java:216)
     [java] 	at org.apache.tools.ant.taskdefs.ExecuteJava.execute(ExecuteJava.java:155)
     [java] 	... 50 more
[ERROR]      [java] Java Result: -1
[ERROR] [replaceregexp] The following file is missing: '/home/suleiman/git/birt/build/org.eclipse.birt.api/target/manifest.mf'
[DEBUG] 
build-api-package:
[DEBUG]       [ant] Exiting /home/suleiman/git/birt/build/org.eclipse.birt.api/build.xml.

**Reply by merks** [reply]
If you look here, you can see how BIRT is built:

https://ci.eclipse.org/birt/job/build/job/master/lastBuild/console

You can set up a development environment like this:

https://github.com/eclipse-birt/birt?tab=readme-ov-file#create-a-birt-development-environment

You kick off a build locally in that IDE:

![image](https://github.com/user-attachments/assets/91e11591-d6d8-4e24-9878-bb59a31377c6)

Of course in that IDE, all the project will be compiled and you can run/test them.

**Comment by dm-ion** [comment]
@sbatyrov I believe that issue is related to JDK 21, try building with JDK 17 (the supported one from merks link) instead.
